src/Connectors/PhpRedisConnector.php

```diff
@@ -0,0 +1,260 @@
+        }
+
+        // Try to connect to any of the servers.
+        foreach ($servers as $idx => $server) {
+            $host = isset($server['host']) ? $server['host'] : 'localhost';
+            $port = isset($server['port']) ? $server['port'] : 26739;
+
+            // Create a connection to the Sentinel instance.
+            $sentinel = new RedisSentinel($host, $port, $timeout, $persistent, $retryWait, $readTimeout);
+
+            try {
+                // Check if the Sentinel server list needs to be updated.
+                // Put the current server and the other sentinels in the server list.
+                $updateSentinels = isset($options['update_sentinels']) ? $options['update_sentinels'] : false;
+                if ($updateSentinels === true) {
+                    $this->updateSentinels($sentinel, $host, $port, $service);
+                }
+
+                // Lookup the master node.
+                $master = $sentinel->getMasterAddrByName($service);
+                if (is_array($master) && ! count($master)) {
+                    throw new RedisException(sprintf('No master found for service "%s".', $service));
+                }
+
+                // Create a PhpRedis client for the selected master node.
+                return $this->createClient(array_merge($parameters, $server, [
+                    'host' => $master[0],
+                    'port' => $master[1],
+                ]));
+            } catch (RedisException $e) {
+                // Rethrow the expection when the last server is reached.
+                if ($idx === count($servers) - 1) {
+                    throw $e;
+                }
+            }
+        }
+    }
+
+    /**
+     * Retry the callback when a RedisException is catched.
+     *
+     * @param callable $callback The operation to execute.
+     * @return mixed The result of the first successful attempt.
+     *
+     * @throws RedisRetryException After exhausting the allowed number of
+     * attempts to connect.
+     */
+    protected function retryOnFailure(callable $callback)
+    {
+        $attempts = 0;
+        $previous = null;
+
+        do {
+            try {
+                return $callback();
+            } catch (RedisException $exception) {
+                $previous = $exception;
+
+                usleep($this->retryWait * 1000);
+
+                $attempts++;
+            }
+        } while ($attempts < $this->retryLimit);
+
+        throw new RedisRetryException(sprintf('Reached the connect limit of %d attempts', $attempts), 0, $previous);
+    }
+
+    /**
+     * Update the list With sentinel servers.
+     *
+     * @param RedisSentinel $sentinel
+     * @param string $currentHost
+     * @param int $currentPort
+     * @param string $service
+     * @return void
+     */
+    protected function updateSentinels(RedisSentinel $sentinel, string $currentHost, int $currentPort, string $service)
+    {
+        $this->servers = array_merge(
+            [
```

Comment:
Shouldn't this opening tag be on the previous line?

Reply:
Ran the specifc code through phpcsfixer, they think its fine. It also makes the merge more readable, especially for the final );